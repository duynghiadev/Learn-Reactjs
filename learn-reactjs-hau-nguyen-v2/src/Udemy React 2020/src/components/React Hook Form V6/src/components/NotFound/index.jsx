import React from 'react';
import PropTypes from 'prop-types';
import './styles.scss';

NotFound.propTypes = {
    
};

function NotFound(props) {
    return (
        <div>
                 Not Found
        </div>
    );
}

export default NotFound;