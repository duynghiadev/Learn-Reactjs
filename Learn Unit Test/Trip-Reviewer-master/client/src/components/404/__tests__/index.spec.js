import React from 'react';
import { create } from 'react-test-renderer';
import { shallow } from 'enzyme';

import NotFound from '..';

const mockComponent = props => <NotFound {...props} />;

describe('<NotFound />', () => {
  it('should render component', () => {
    const tree = create(mockComponent()).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should redirect to home after a click on the button', () => {
    const push = jest.fn();
    const props = { history: { push } };
    const node = shallow(mockComponent(props));

    node.find('.button').simulate('click');

    expect(push).toHaveBeenCalledWith('/');
  });
});
