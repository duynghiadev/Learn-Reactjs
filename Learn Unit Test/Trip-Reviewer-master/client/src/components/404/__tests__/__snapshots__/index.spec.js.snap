// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NotFound /> should render component 1`] = `
<div
  className="not-found-container"
>
  <h1
    className="title"
  >
    404
  </h1>
  <div
    className="picture"
  />
  <h2
    className="text"
  >
    It looks like you're lost...
  </h2>
  <button
    className="button"
    onClick={[Function]}
  >
    Go Home
  </button>
</div>
`;
